peewee.py

```diff
@@ -4401,6 +4402,21 @@ def __init__(self, *args, **kwargs):
             setattr(self, k, v)
 
     @classmethod
+    def get_read_database(self):
+        return self._meta.database
+
+    @classmethod
+    def get_write_database(self):
+        return self._meta.database
+
+    @classmethod
+    def determine_database(self, query, sql, params):
+        if sql.lower().startswith('select'):
+            return self.get_read_database()
+        else:
+            return self.get_write_database()
```

Comment:
I am concerned about negative performance implications with this code. Perhaps a better option would be to move this type of logic into the `read_slave` module. Although honestly, the best solution IMO would be a separate third party package.